kernel/packages/unity-interface/UnityInterface.ts

```diff
@@ -188,6 +190,28 @@ export class UnityInterface {
     this.SendMessageToUnity('Main', 'SetDisableAssetBundles')
   }
 
+  public async CrashPayloadRequest(): Promise<string> {
+    // Over wasm this should come back on the same call stack frame because
+    // the response comes within the CrashPayloadRequest method body.
+
+    // For websocket this should take more frames, so we need promises.
+    let promise = new Promise<string>((resolve, reject) => {
+      let crashListener = this.crashPayloadResponseObservable.addOnce((payload) => {
+        resolve(payload)
+      })
+
+      // We solve on timeout anyways to simplify usage.
+      setTimeout(() => {
+        this.crashPayloadResponseObservable.remove(crashListener)
+        resolve('Crash payload request failed')
```

Comment:
why resolving instead of rejecting here?

Reply:
Check this comment:

`We solve on timeout anyways to simplify usage.`  

This means the failure doesn't need to be handled, we'd just track an error with an empty payload if this happens.